compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -7245,7 +7246,8 @@ private BType defineInferredRecordType(BLangRecordLiteral recordLiteral, BType e
     private BRecordTypeSymbol createRecordTypeSymbol(PackageID pkgID, Location location,
                                                      SymbolOrigin origin) {
         BRecordTypeSymbol recordSymbol =
-                Symbols.createRecordSymbol(0, names.fromString(anonymousModelHelper.getNextAnonymousTypeKey(pkgID)),
+                Symbols.createRecordSymbol(Flags.ANONYMOUS,
```

Comment:
Looking at the usage of `org.wso2.ballerinalang.compiler.semantics.model.symbols.Symbols#createRecordSymbol`, there seem to be some other places where we create the symbol without the `Flags.ANONYMOUS` flag. Don't we have to change them too?

Reply:
Ah yes, will fix those as well.